common/src/main/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/AuditRecordWriteMarshallable.java

```diff
@@ -29,9 +29,18 @@
     public AuditRecordWriteMarshallable(AuditRecord auditRecord, FieldSelector configuredFields)
     {
         this.auditRecord = auditRecord;
-        this.actualFields = auditRecord.getBatchId().isPresent()
-                            ? configuredFields
-                            : configuredFields.withoutField(Field.BATCH_ID);
+        this.actualFields = getFieldsAvailableInRecord(auditRecord, configuredFields);
+    }
+
+    final static FieldSelector getFieldsAvailableInRecord(AuditRecord auditRecord, FieldSelector configuredFields)
+    {
+        FieldSelector fields = auditRecord.getBatchId().isPresent()
+                               ? configuredFields
+                               : configuredFields.withoutField(Field.BATCH_ID);
+
+        return auditRecord.getClientAddress() != null
```

Comment:
Hmm, perhaps we agreed on somethign else when we discussed this earlier.

But I'm thinking that we shouldn't merge this forward. Reason is that if (when) we drop support for 2.2, we will still have this code in all branches for no good reason.

Reply:
Ok, I'll remove this change...